### PR TITLE
Switch Linux CI to jammy

### PR DESCRIPTION
**Pull request by veloman-yunkan**
Fixed the Windows CI (broken because of kiwix/kiwix-build#779) and swtiched the Linux CI to jammy.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,16 +56,16 @@ jobs:
...
-            image_variant: focal
+          - target: jammy
+            image_variant: jammy
             lib_postfix: '/x86_64-linux-gnu'
     env:
       HOME: /home/runner
     runs-on: ubuntu-22.04
     container:
-      image: ghcr.io/kiwix/kiwix-build_ci_${{matrix.image_variant}}:38
+      image: ghcr.io/kiwix/kiwix-build_ci_${{matrix.image_variant}}:latest
```
Comment (kelson42) [inline .github/workflows/ci.yml:68]: Please used here a pined version. If I'm not mistaken, this should be `2024-11-06`

**Review by kelson42 — CHANGES_REQUESTED** [review]
See my comment and we are good to me.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (veloman-yunkan) [reply .github/workflows/ci.yml:68]: There is no such tag for `https://ghcr.io/kiwix/kiwix-build_ci_focal`

Comment (veloman-yunkan) [reply .github/workflows/ci.yml:68]: Oops. I confused this PR with kiwix/kiwix-build#771

Comment (veloman-yunkan) [reply .github/workflows/ci.yml]: Done

**Pull request merged**